### ISSUE TITLE
Implemented operations cannot be edited after changing API definition

### ISSUE THREAD

**Issue by mmuzikar** [syndesisio/syndesis#6099]
## This is a...

<pre><code>
[ ] Feature request
[x] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description
If I implement an operation and "edit" the definition, all implemented operations appear as if they were not implemented.
**Only way to edit those operations is by canceling the operation edit process and edit it again.**

![Peek 2019-07-09 12-20](https://user-images.githubusercontent.com/46345469/60880820-af0f4480-a244-11e9-8cf2-e88200a17065.gif)

**Comment by zregvart** [comment]
Looks like the editing of the OpenAPI definition switched the flow of the logic to the flow for the new integration, and in the course of that lost the old flow definitions.

**Comment by riccardo-forina** [comment]
@zregvart is the API that returns the updated integration keeping the `metadata` values set in each flow?

**Comment by zregvart** [comment]
Looking at this:

https://github.com/syndesisio/syndesis/blob/610a530fe375998afc4736e10aa866017678bdd9/app/server/endpoint/src/main/java/io/syndesis/server/endpoint/v1/handler/api/ApiGeneratorHelper.java#L145

Only the start/end step data shapes should change.

**Comment by asmigala** [comment]
I consider this a p0, as it loses user input.

**Comment by zregvart** [comment]
I've deployed 1.7.9 to minishift it seems that the update via `PUT /api/v1/apis/generator` is missing the initial flow metadata received from `POST /api/v1/apis/generator`. In that case the OpenAPI operation can't be matched to a flow because it's missing the `openapi-operationid` from the flow metadata and the server decides to remove the implemented flow and create a new flow for the same operation.

**Comment by heiko-braun** [comment]
@riccardo-forina @zregvart Do we already have an idea what it takes to fix this?

**Comment by heiko-braun** [comment]
@riccardo-forina What do you think about this part in the description?

> Only way to edit those operations is by canceling the operation edit process and edit it again.

**Comment by zregvart** [comment]
I was wrong on https://github.com/syndesisio/syndesis/issues/6099#issuecomment-510420382 I think I was looking at the wrong `metadata` in the integration JSON.

**Comment by riccardo-forina** [comment]
> @riccardo-forina What do you think about this part in the issue description?
> 
> > Only way to edit those operations is by canceling the operation edit process and edit it again.

I don't know, Zoran and I can not reproduce this either on master nor on 1.7. Our experience is that all changes are preserved, and the UI shows the edited operations as edited. So how it should behave.

**Comment by zregvart** [comment]
I can't reproduce this as shown in the GIF from the initial report. I've installed Syndesis 1.7.9 with `syndesis minishift --install --tag 1.7.9` (about says dec6c1793b1a5804ef2c0e5ab7b954bf1fa51ac5). I think I followed the steps exactly and used the same OpenAPI starting document.

![Peek 2019-07-12 09-59](https://user-images.githubusercontent.com/1306050/61112418-0e12ca80-a48c-11e9-9592-3b4ba45b6f0e.gif)

The only odd bit was that the integration name switched to `New integration`, so other than needing to specify the name when saving the integration no other side effects that I can see.

What am I missing?
